src/something_main.cpp

```diff
@@ -154,9 +154,22 @@ int main(void)
 
     Uint32 prev_ticks = SDL_GetTicks();
     float lag_sec = 0;
+    float next_sec = 0;
+    size_t frames_of_current_second = 0;
+    size_t fps = 0;
     while (!game.quit) {
         Uint32 curr_ticks = SDL_GetTicks();
         float elapsed_sec = (float) (curr_ticks - prev_ticks) / 1000.0f;
+
+        frames_of_current_second += 1;
+        next_sec += elapsed_sec;
+
+        if (next_sec >= 1) {
+            fps = frames_of_current_second;
+            next_sec = 0;
```

Comment:
```suggestion
            next_sec -= 1.0f;
```
`next_sec` is float, and mostly it's not exactly equal to 1.0f. Thus simple value reset to 0 makes interval of fps update greater than 1 s. Without this fix, on my machine I see fps = 61.

Reply:
If you do -1s let me show you the table and the issue
| actual_sec | next_sec | will display  |  fps (with your solution) |
|---|---|---|---|
| 0.7  | 0.7 | No | 0 |
| 1.4  | 0.4 | Yes | 2 | 
| 2.1  | 0.1 | Yes | 1 |
| 2.8 | 0.8 | No | 1 |
| 3.5 | 0.5 | Yes |  2 |

With my code it will always show 2. since 0.7 -> 0.0 -> 0.7 -> 0.0 -> 0.7 etc

Reply:
But with constant `dt = 0.7` we have `FPS = 1.42`. So 1 and 2 are valid FPS to display. Here's [interactive spreadsheet](https://docs.google.com/spreadsheets/d/1uI9e_o9V0IjaggzkLElW7JaWc164ngfYUAjHyt3HgDA/edit?usp=sharing), where you can paste different dt in orange cell.

Reply:
I'm sorry, I did what you said first and then I managed to mindfuck myself and changed the solution I had.